src/CommunityToolkit.Maui/Essentials/FolderPicker/FolderPickerImplementation.android.cs

```diff
@@ -0,0 +1,41 @@
+using CommunityToolkit.Maui.Alerts;
+using CommunityToolkit.Maui.Core;
+using CommunityToolkit.Maui.Core.Primitives;
+
+namespace CommunityToolkit.Maui.Storage;
+
+/// <inheritdoc />
+public class FolderPickerImplementation : IFolderPicker
+{
+	/// <inheritdoc />
+	public async Task<Folder> PickAsync(string initialPath, CancellationToken cancellationToken)
+	{
+		var status = await Permissions.RequestAsync<Permissions.StorageRead>();
+		if (status is not PermissionStatus.Granted)
+		{
+			throw new PermissionException("Storage permission is not granted");
+		}
+
+		var currentActivity = Platform.CurrentActivity ?? throw new InvalidOperationException($"{nameof(Platform.CurrentActivity)} cannot be null");
+		var dialog = new FileFolderDialog(currentActivity, FileSelectionMode.FolderChoose);
+		var path = await dialog.GetFileOrDirectoryAsync(initialPath, cancellationToken: cancellationToken);
+
+		return new Folder
+		{
+			Path = path,
+			Name = new DirectoryInfo(path).Name
+		};
+	}
+
+	/// <inheritdoc />
+	public Task<Folder> PickAsync(CancellationToken cancellationToken)
+	{
+		return PickAsync(GetExternalDirectory(), cancellationToken);
+	}
+
+	static string GetExternalDirectory()
+	{
+		return Platform.CurrentActivity?.GetExternalFilesDir(null)?.ParentFile?.ParentFile?.ParentFile?.ParentFile?.AbsolutePath
+				?? "/storage/emulated/0";
```

Comment:
this `"/storage/emulated/0"` is always safe? Would be better to ask the user to provide a default path?

Reply:
it is only used in case user wants to use default location (see line 44). We also have a method where user can set his own path